scipy/spatial/distance.py

```diff
@@ -360,6 +360,12 @@ def correlation(u, v):
     """
     u = _validate_vector(u)
     v = _validate_vector(v)
+
+    # Deal with the corner cases at first
+    # Avoid to output nan in any case
+    if np.ptp(u) == 0.0 or np.ptp(v) == 0.0:
```

Comment:
You are iterating over each array one more time by calling `ptp` on them.  Wouldn't it be better to store `norm(um)` and `norm(vm)`, and do the check on them?  Also, if both vectors are constant, is the correlation still 0 or is it 1 then?